api/src/middleware/competenciesRouter.ts

```diff
@@ -48,4 +50,47 @@ competenciesRouter.post('/', async (req, res, next) => {
   res.status(201).json(itemEnvelope(competency));
 });
 
+competenciesRouter.put('/:id', async (req, res, next) => {
+  const { principalId } = req.session;
+  const { id } = req.params;
+
+  const competencyId = Number(id);
+  if (!Number.isInteger(competencyId) || competencyId < 1) {
+    next(new BadRequestError(`"${id}" is not a valid competency id.`));
+    return;
+  }
+
+  let isAuthorized;
+  try {
+    isAuthorized = await authorizeCompetencyUpdate(principalId, competencyId);
+  } catch (error) {
+    next(error);
+    return;
+  }
+  if (!isAuthorized) {
+    next(
+      new NotFoundError(
+        `A competency with the id "${competencyId}" could not be found.`
+      )
+    );
+    return;
+  }
+  const { label, description } = req.body;
```

Comment:
I think this declaration should be at the top of the function.

Reply:
Now that we have blocked scoped variable declarations and we don't have to worry about hoisting, we can define variables as close to their first use as possible in order to make the code more readable and maintainable.

Reply:
Thanks for the explanation.